### PR TITLE
fix(rollbacks): support for tolerating some instance failures

### PR DESCRIPTION
**Pull request by asher**
[AWS] Allow rollbacks to proceed when < 100% of instances in the rollback asg report healthy.

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy [hunk 1]
```diff
@@ -186,8 +188,10 @@ abstract class AbstractEnableDisableAtomicOperation implements AtomicOperation<V
             credentials: credentials,
             region: region,
             asgName: serverGroupName,
-            instanceIds: instanceIds
+            instanceIds: instanceIds,
+            targetHealthyDeployPercentage: description.targetHealthyDeployPercentage ?: 100
         )
+        // @TODO pass targetHealthy% here
```
Comment (cfieber) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy]: what was the TODO here?

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy [hunk 2]
```diff
@@ -186,8 +188,10 @@ abstract class AbstractEnableDisableAtomicOperation implements AtomicOperation<V
             credentials: credentials,
             region: region,
             asgName: serverGroupName,
-            instanceIds: instanceIds
+            instanceIds: instanceIds,
+            targetHealthyDeployPercentage: description.targetHealthyDeployPercentage ?: 100
```
Comment (cfieber) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy]: I think (not totally sure) that the groovy truthiness here would convert an explicitly supplied value of `0` to `100`

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy (hunk 1 above)
Comment (asher) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy]: It was resolved by adding targetHealthyDeployPercentage to EnableDisableAsgDescription && AbstractRegionAsgInstanceIdsDescription, removing the TODO.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy (hunk 2 above)
Comment (asher) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AbstractEnableDisableAtomicOperation.groovy]: Verified that you're right - going to change this to the more explicit `description.targetHealthyDeployPercentage != null ? description.targetHealthyDeployPercentage : 100`

**Pull request merged**